ts/src/derive.ts

```diff
@@ -0,0 +1,2594 @@
+        const prefix = this.base16ToBinary ('1901');
+        return this.hash (this.binaryConcat (prefix, binaryDomainSeparator, accountHash), keccak, 'hex');
+    }
+
+    signOrder (order, privateKey) {
+        return this.signHash (this.hashOrderMessage (order).slice (-64), privateKey.slice (-64));
+    }
+
+    hashMessage (message) {
+        // takes a utf8 encoded message
+        const binaryMessage = this.encode (message);
+        const binaryMessageLength = binaryMessage.length;
+        const prefix = this.encode ('\x19Ethereum Signed Message:\n' + this.numberToString (binaryMessageLength));
+        return '0x' + this.hash (this.binaryConcat (prefix, binaryMessage), keccak, 'hex');
+    }
+
+    signHash (hash, privateKey) {
+        const signature = ecdsa (hash.slice (-64), privateKey.slice (-64), secp256k1, undefined);
+        return '0x' + signature['r'].padStart (64, '0') + signature['s'].padStart (64, '0') + this.intToBase16 (this.sum (27, signature['v']));
+    }
+
+    signMessage (message, privateKey) {
+        return this.signHash (this.hashMessage (message), privateKey.slice (-64));
+    }
+
+    parseUnits (num: string, dec = '1000000000000000000') {
+        return Precise.stringMul (num, dec);
+    }
+
+    /**
+     * @method
+     * @name derive#createOrder
+     * @description create a trade order
+     * @see https://docs.derive.xyz/reference/post_private-order
+     * @param {string} symbol unified symbol of the market to create an order in
+     * @param {string} type 'market' or 'limit'
+     * @param {string} side 'buy' or 'sell'
+     * @param {float} amount how much of currency you want to trade in units of base currency
+     * @param {float} [price] the price at which the order is to be fulfilled, in units of the quote currency, ignored in market orders
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @param {float} [params.triggerPrice] The price a trigger order is triggered at
+     * @param {object} [params.takeProfit] *takeProfit object in params* containing the triggerPrice at which the attached take profit order will be triggered (perpetual swap markets only)
+     * @param {float} [params.takeProfit.triggerPrice] take profit trigger price
+     * @param {object} [params.stopLoss] *stopLoss object in params* containing the triggerPrice at which the attached stop loss order will be triggered (perpetual swap markets only)
+     * @param {float} [params.stopLoss.triggerPrice] stop loss trigger price
+     * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+     */
+    async createOrder (symbol: string, type: OrderType, side: OrderSide, amount: number, price: Num = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        let subaccountId = undefined;
+        [ subaccountId, params ] = this.handleOptionAndParams (params, 'createOrder', 'subaccount_id');
```

Comment:
@sc0Vu is subaccount mandatory? in that case why are we  not throwing when not available?

Reply:
@sc0Vu id price is required, why aren't we explicit about it.

Besides that, as we mentioned multiple times `toString` shouldn't be user to convert numbers to string, you should use `numberToString` instead